### PR TITLE
Add Rust-Diesel example

### PR DESCRIPTION
**Pull request by svranesevic**

### CODE REVIEW

**Comment by sibtayshah** [comment]
Please remove Cargo.lock file

**Comment by svranesevic** [comment]
I would suggest that `Cargo.lock` is kept. (See https://github.com/rust-lang/cargo/issues/315#issuecomment-51075730)
Let me know what you think after checking out attached post

File: rust/diesel/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+DATABASE_URL=postgres://postgres:mysecretpassword@localhost/yugabyte_demo
```
Comment (sibtayshah) [inline rust/diesel/.env]: database should be ysql_diesel. Password should be null

File: rust/diesel/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Build and run
+
+Build the REST API server (written using Diesel and Rocket) as follows:
+
+```
+$ cargo build --release
```
Comment (sibtayshah) [inline rust/diesel/README.md:6]: is this also going to create the DB schema?

Comment (svranesevic) [reply rust/diesel/README.md:6]: Yes, the DB schema is updated/created on application startup

**Comment by sibtayshah** [comment]
ok lets add Cargo.lock

**Pull request merged**